mmv1/provider/terraform_kcc.rb

```diff
@@ -17,25 +17,64 @@
 module Provider
   # Magic Modules Provider for KCC ServiceMappings and TF samples.
   class TerraformKCC < Provider::Terraform
+    PRODUCT_NAME_MAP = { Alloydb: 'AlloyDB',
+                         ApiGateway: 'APIGateway',
+                         Beyondcorp: 'BeyondCorp',
+                         BigqueryAnalyticsHub: 'BigQueryAnalyticsHub',
+                         BigqueryConnection: 'BigQueryConnection',
+                         BigqueryDatapolicy: 'BigQueryDataPolicy',
+                         BigqueryDataTransfer: 'BigQueryDataTransfer',
+                         BigqueryReservation: 'BigQueryReservation',
+                         CloudIds: 'CloudIDS',
+                         CloudIot: 'CloudIOT',
+                         Pubsub: 'PubSub' }.freeze
+    OBJECT_NAME_MAP = { Api: 'API',
+                        Dns: 'DNS',
+                        Dicom: 'DICOM',
+                        Entitytype: 'EntityType',
+                        Fhir: 'FHIR',
+                        Gcp: 'GCP',
+                        Hl7: 'HL7',
+                        Hmac: 'HMAC',
+                        Idp: 'IDP',
+                        Nat: 'NAT',
+                        Saml: 'SAML',
+                        Ssl: 'SSL',
+                        Url: 'URL' }.freeze
+
     def generate(output_folder, types, _product_path, _dump_yaml, generate_code, generate_docs)
       @base_url = @version.base_url
       generate_objects(output_folder, types, generate_code, generate_docs)
       compile_product_files(output_folder)
     end
 
+    def generate_product_name(product_name)
```

Comment:
nit: I would lean away from `generate_` as a prefix for a function that's not generating a file. Not a big deal, though.